libctl/structure.cpp

```diff
@@ -1300,8 +1297,18 @@ static meep::susceptibility *make_multilevel_sus(const multilevel_atom *d) {
   for (int t = 0; t < d->transitions.num_items; ++t) {
     int i = d->transitions.items[t].from_level - minlev;
     int j = d->transitions.items[t].to_level - minlev;
-    Gamma[i*L+i] += d->transitions.items[t].transition_rate;
-    Gamma[j*L+i] -= d->transitions.items[t].transition_rate;
+    Gamma[i*L+j+1] -= d->transitions.items[t].transition_rate;
+    Gamma[(i-1)*L+j+1] += d->transitions.items[t].transition_rate;
+  }
+  
+  // pumping rate matrix Rp
+  meep::realnum *Rp = new meep::realnum[L * L];
+  memset(Rp, 0, sizeof(meep::realnum) * (L*L));
+  for (int t = 0; t < d->transitions.num_items; ++t) {
+    int i = d->transitions.items[t].from_level - minlev;
+    int j = d->transitions.items[t].to_level - minlev;
+    Rp[i*L+i] -= d->transitions.items[t].pumping_rate;
```

Comment:
The sign convention of the entries of the pumping-rate matrix Rp are negative for the lower state (negative pump) and positive for the higher one. This is the same convention used in Song's paper and is also consistent with the way this term should be (and is) defined in the interface: from lower state to higher state. This sign convention for the Gamma matrix is flipped since the non-radiative decay is from the higher state to the lower state which is again how it should be and is defined in the interface. The convention that you used in the original formulation of the Gamma matrix had this reversed.